src/Container/Context.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Container;
+
+use ReflectionClass;
+use ReflectionFunction;
+use ReflectionMethod;
+
+final class Context
```

Comment:
Can this be `readonly` to prevent the modification of the class properties (if they need to be exposed)?

Reply:
Technically, it can, because you can add items to readonly arrays. But I find that confusing. A property is readonly, but you can still change its contents. That's why I prefer non-readonly properties in those cases.

However, these should just be private, I think